docs/reference/aggregations/metrics/scripted-metric-aggregation.asciidoc

```diff
@@ -278,6 +278,11 @@ params::           Optional. An object whose contents will be passed as variable
 --------------------------------------------------
 // NOTCONSOLE
 
+needs_scores::     Optional, defaults to `true`.  By default, scripts are able to access scores, but this is incompatible with some
+                   aggregations, notably nested composites.  Setting this to `false` indicates that the script is not using scores, and can
+                   be run in a mode that doesn't support scoring.  Note that no attempt is made to validate the script against this setting,
+                   and the behavior of accessing scores while `needs_scores` is set to `false` is undefined.
+
```

Comment:
We have the ability to detect whether aggregation scripts need scores (see e.g. `SumAggregator` which delegates to the values source, wich then delegates to the script). Would it be possible to do the same for `map` scripts instead of requiring users to tell us?

Reply:
It doesn't look possible to me.  Or, well, it doesn't look like the `MapScript` already has the ability to analyze the script and see if it needs scores.  `SumAggregator` does indeed get that from the `ValuesSource`, which get it from the script, but the script in that case is an `AggregationScript`, which implements a `needsScores()` method.  `MapScript` descends from `DocBasedScript`, which does not define a `needsScores()` method.  There's a TODO in scripted metric to implement such a thing.

I'm not opposed to building that, but it seemed like more work than adding a parameter, and we have been hesitant to invest much into scripted metric recently.